watertap/core/util/scaling.py

```diff
@@ -0,0 +1,96 @@
+###############################################################################
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+"""
+This module contains utility functions for model statistics of WaterTAP models.
+"""
+
+import pyomo.environ as pyo
+from pyomo.environ import check_optimal_termination
+from pyomo.common.collections import ComponentSet
+
+import idaes.logger
+from idaes.core.solvers import get_solver
+from idaes.core.util.scaling import (
+    calculate_scaling_factors,
+    badly_scaled_var_generator,
+    get_scaling_factor,
+    unset_scaling_factor,
+)
+
+solver = get_solver()
+
+__author__ = "Hunter Barber"
+
+
+def variable_sens_generator(blk, lb_scale=1e-1, ub_scale=1e3, tol=1e2, zero=1e-10):
+    """
+
+    Keyword Arguments:
+
+    Returns:
+
+    """
```

Comment:
Thanks for taking the initiative on this. I think once you document this more clearly via docstrings, add logging (similar to what's done in `watertap/core/util/infeasible` and `watertap/core/util/initialization`), and then add dedicated testing to verify the intended functionality of this tool, it will probably be a helpful addition to our utilities.